src/dmd/e2ir.d

```diff
@@ -514,7 +514,19 @@ if (!irs.params.is64bit) assert(tysize(TYnptr) == 4);
             e.Eflags |= EFLAGS_variadic;
     }
 
-    if (retmethod == RET.stack)
+    const isCPPCtor = fd && fd.linkage == LINK.cpp && fd.isCtorDeclaration();
+    if (isCPPCtor &&
+        (irs.params.isLinux ||
+         irs.params.isOSX ||
+         irs.params.isFreeBSD ||
+         irs.params.isDragonFlyBSD ||
+         irs.params.isSolaris))
```

Comment:
Perhaps some `isPOSIX` function would be better suited here. The C++ side has [this](https://github.com/dlang/dmd/blob/9590dc86eca66e55cf9e28ca600e69589ff2ac35/src/dmd/root/object.h#L12).